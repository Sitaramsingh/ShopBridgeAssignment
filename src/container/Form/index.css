.formStyle{
    box-shadow: 0px 3px 6px #00000029;
    padding: 10px;
    display: flex;
    flex-wrap: wrap;
}
.formStyle  div div input::-webkit-inner-spin-button {
    -webkit-appearance: none;
  }
.flexInner{
    flex: 100%;
}
.flexInner div{
    width: 100%;
}
.desflex{
    flex: 100%;
}
.desflex div{
    width: 100%;
}
.modalStyle{
   min-width: 200px; 
}
.btnContainer{
    margin: auto;
}

@media(min-width: 767px){
    .modalStyle{
      min-width: 600px;
      left: 25%; 
    }
    .flexInner{
        flex: 35%;
    }
}