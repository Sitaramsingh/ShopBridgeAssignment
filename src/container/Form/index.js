import React,{useState, useEffect} from 'react';
import InputBox from  '../../component/InputBox';
import Button from '../../component/Button';
import Modal from '../../component/modal';
import './index.css';

import {themes, nameConstants} from '../../constant';
// import {addItemAction, editItemAction} from '../../action';

const createForm = [{text:'title', label: 'title', placeHolder: 'Enter title', type:'text'}, {text:'name', label:'Name', placeHolder: 'Enter Name',  type:'text'}, {text:'description', label:'Description', placeHolder: 'Enter Description',  type:'text'},  {text:'offer', label:'offer', placeHolder: 'Enter offer',  type:'text'},{text:'price',label: 'Price', placeHolder: 'Enter Price',  type:'number'}];

export default function Index(props) {
    const [state, setState] = useState({
        id: '',
        title: '',
        name: '',
        description: '',
        price: '',
        offer: ''
    });

    const {editableItem} = props;

    const handleChange = (evt) => {
        const name = evt.target.name;
        const value = evt.target.value;
        setState({ ...state,  [name]: value });
    }

    const onSubmitItems = (evt) => {
        evt.preventDefault();
        const reqPayload = state;
        props.formSubmit(reqPayload);
        props.handleFormModal();

    }
    useEffect(() => {
        if(editableItem){
            const {name, description, price, id, offer, title} = editableItem;
            setState(prevState => {
                return {
                    ...prevState,
                    name: name,
                    description: description,
                    price: price,
                    id: id,
                    title: title,
                    offer: offer
                };
            });
        }
    }, [editableItem])

    const handleValidation  = () => {
        if(state.title && state.name && state.description && state.price && state.offer){
            return false
        }else {
            return true
        }

    }

    const onCancel = (evt) => {
        evt.preventDefault();
        props.handleFormModal();
        
    }

    return (
        <Modal open={props.isAddItem} onclose={props.handleFormModal} className='modalStyle'>
            <div>
                <h1 className={'mt-4'}>{state.id ? nameConstants.UPDATEPRODUCT : nameConstants.ADDPRODUCT}</h1>
                <form className='pd-2 formStyle'>
                    {createForm.map((item, index)=> <div className= {['m-2',  item.text === 'description' ? 'desflex':'flexInner'].join(' ')}  key={index.toString()}><InputBox variant='outlined'  type={item.type} label={item.label} key={item.text} name={item.text} value={state[item.text]} placeholder={item.placeHolder} onChange={handleChange}/></div>)}
                    <div className='btnContainer'>
                        <Button className={['m-2' ,themes.SECONDARY_BUTTON].join(' ')} onClick={onCancel} > Cancel </Button>
                        <Button  className={themes.PRIMARY_BUTTON} disable = {handleValidation()} onClick={onSubmitItems}> {state.id ?  'Update' :'Save'} </Button>
                    </div>
                </form>
            </div>
        </Modal>
    );
}
