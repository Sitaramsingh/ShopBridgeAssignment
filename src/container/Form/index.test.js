import React from 'react';
import { shallow } from 'enzyme';
import Form from './index';

describe('Component should be render ', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Form/>);
    expect(component).toMatchSnapshot();
  });
});