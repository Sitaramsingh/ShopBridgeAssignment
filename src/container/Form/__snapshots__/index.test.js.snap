// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should be render  should render correctly in "debug" mode 1`] = `
<modal>
  <div>
    <h1
      className="mt-4"
    >
      Add Prodcuts
    </h1>
    <form
      className="pd-2 formStyle"
    >
      <div
        className="m-2"
      >
        <inputType
          key="name"
          label="Name"
          name="name"
          onChange={[Function]}
          placeholder="Enter Name"
          value=""
          variant="outlined"
        />
      </div>
      <div
        className="m-2"
      >
        <inputType
          key="description"
          label="Description"
          name="description"
          onChange={[Function]}
          placeholder="Enter Description"
          value=""
          variant="outlined"
        />
      </div>
      <div
        className="m-2"
      >
        <inputType
          key="price"
          label="Price"
          name="price"
          onChange={[Function]}
          placeholder="Enter Price"
          value=""
          variant="outlined"
        />
      </div>
      <button
        className="m-2 btn btn-secondary"
        onClick={[Function]}
      >
         Cancel 
      </button>
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
         
        Save
         
      </button>
    </form>
  </div>
</modal>
`;
